apollo-router/tests/integration/subscriptions/ws_passthrough.rs

```diff
@@ -917,3 +918,159 @@ async fn test_subscription_ws_passthrough_dedup() -> Result<(), BoxError> {
+    router.replace_config_string("http://localhost:{{ACCOUNTS_PORT}}", &ws_url);
+    router.replace_config_string("rng:", "accounts:");
+
+    info!("WebSocket server started at: {}", ws_url);
+
+    router.start().await;
+    router.assert_started().await;
+
+    // Use the configured query that matches our server configuration
+    let query = create_sub_query(interval_ms, custom_payloads.len());
+    let ((_, response), (_, response_bis)) = futures::join!(
+        router.run_subscription(&query),
+        router.run_subscription(&query)
+    );
+
+    // Expect the router to handle the subscription successfully
+    assert!(
+        response.status().is_success(),
+        "Subscription request failed with status: {}",
+        response.status()
+    );
+    assert!(
+        response_bis.status().is_success(),
+        "Subscription request failed with status: {}",
+        response_bis.status()
+    );
+
+    let metrics = router.get_metrics_response().await?.text().await?;
+    let sum_metric_counts = |regex: &Regex| {
+        regex
+            .captures_iter(&metrics)
+            .flat_map(|cap| cap.get(1).unwrap().as_str().parse::<usize>())
+            .sum()
+    };
+
+    let stream = response.bytes_stream();
+    let stream_bis = response_bis.bytes_stream();
+
+    let deduplicated_sub =
+        Regex::new(r#"(?m)^apollo_router_operations_subscriptions_total[{].+subscriptions_deduplicated="true".+[}] ([0-9]+)"#)
+            .expect("regex");
+    let total_deduplicated_sub: usize = sum_metric_counts(&deduplicated_sub);
+    assert_eq!(total_deduplicated_sub, 1);
+    let duplicated_sub =
+        Regex::new(r#"(?m)^apollo_router_operations_subscriptions_total[{].+subscriptions_deduplicated="false".+[}] ([0-9]+)"#)
+            .expect("regex");
+    let total_duplicated_sub: usize = sum_metric_counts(&duplicated_sub);
+    assert_eq!(total_duplicated_sub, 1);
+
+    // We'll start consuming both subscriptions, but cancel the first one as soon as a message is
+    // received. the `bis` subscription should continue to receive messages after that.
+    let mut multipart = multer::Multipart::new(stream, "graphql");
+    let mut multipart_bis = multer::Multipart::new(stream_bis, "graphql");
+
+    let task = tokio::task::spawn(tokio::time::timeout(Duration::from_secs(30), async move {
+        let expected_event = create_expected_user_payload(1);
+        while let Some(field) = multipart
+            .next_field()
+            .await
+            .expect("could not read next chunk")
+        {
+            let parsed: serde_json::Value = field.json().await.expect("invalid JSON chunk");
+            // Heartbeat
+            if parsed == serde_json::json!({}) {
+                continue;
+            }
+            assert_eq!(parsed, expected_event);
+            // Close the connection early. the other side should continue to receive events...
```

Comment:
The other side being the other subscription, or the websocket? 

If the websocket, basically we are saying:
* there are two subscriptions running
* both are connected to the same websocket
* if one of them is closed, the websocket should remain open?